File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
import java.util.Arrays;

class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int window_sum = 0;
        int i = 0;
        while(i < k){
            window_sum += nums[i];
            i++;
        }
        double max_average = window_sum / (double) k;
        for(int j = k; j < nums.length; j++){
            window_sum += nums[j] - nums[j - k];
            max_average = Math.max(max_average, window_sum / (double) k);
        }
        return max_average;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        # left = 0
        # current_sum = sum(nums[0:k])
        # max_average = current_sum/k
        # for right in range(k, len(nums)):
        #     current_sum = (current_sum + nums[right] - nums[left])
        #     max_average = max(max_average, current_sum / k)
        #     left += 1

        # return max_average
        window_sum = sum(nums[0:k])
        max_average = window_sum / k

        for i in range(k, len(nums)):
            window_sum += nums[i] - nums[i - k]
            max_average = max(max_average, window_sum / k)
        
        return max_average